go/libkb/stream_classifier.go

```diff
@@ -235,7 +235,8 @@ func ClassifyStream(r io.Reader) (sc StreamClassification, out io.Reader, err er
 	peeker := NewStreamPeeker(r)
 	var buf [100]byte
 	var n int
-	if n, err = peeker.Peek(buf[:]); err != nil {
+	// ErrUnexpectedEOF might just mean we read less than 100 bytes
+	if n, err = peeker.Peek(buf[:]); err != nil && !(err == io.ErrUnexpectedEOF && len(buf) > 0) {
```

Comment:
UnknownStreamError gets returned anyway in the short message detached case, but yeah, I need to do that for the other cases.